packages/imx-wallet/src/domain.ts

```diff
@@ -1,8 +1,14 @@
+import type { Link } from "@imtbl/imx-sdk"
+
 export type ImxNetwork = "mainnet" | "ropsten"
 export type ImxEnv = "dev" | "prod"
 
-export type ImxWalletProviderName = "METAMASK" | "MAGIC_LINK" | "WALLET_CONNECT" | "NONE"
-
+export enum ImxWalletProviderEnum {
```

Comment:
Lets use enums for typescript users
it's more friendly way and have fallbacks for strings